addons/stock/models/stock_warehouse.py

```diff
@@ -86,6 +86,7 @@ def _default_name(self):
     resupply_route_ids = fields.One2many(
         'stock.route', 'supplied_wh_id', 'Resupply Routes',
         help="Routes will be created for these resupply warehouses and you can select them on products and product categories")
+    user_ids = fields.One2many('res.users', 'property_warehouse_id', 'Users')
```

Comment:
It seems to work fine on my end. Since property_warehouse_id is company_dependent, using the filters in another company with no default warehouse simply returns nothing.